app/collins/controllers/actions/ipmi/GetPoolsAction.scala

```diff
@@ -0,0 +1,114 @@
+package collins.controllers.actions.ipmi
+
+import scala.concurrent.Future
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+
+import collins.controllers.ResponseData
+import collins.controllers.SecureController
+import collins.controllers.actions.RequestDataHolder
+import collins.controllers.actions.SecureAction
+import collins.controllers.actions.ipaddress.AddressActionHelper
+import collins.models.IpmiInfo
+import collins.models.shared.AddressPool
+import collins.util.security.SecuritySpecification
+
+case class GetPoolsAction(
+  spec: SecuritySpecification,
+  handler: SecureController
+) extends SecureAction(spec, handler) with AddressActionHelper {
+
+  case class ActionDataHolder(pools: Set[String]) extends RequestDataHolder
+
+  override def validate(): Validation = {
+    var pools: Set[String] = {
+      val p = IpmiInfo.AddressConfig.filter(_.poolNames != Set())
+      if (p.isDefined) {
+        IpmiInfo.AddressConfig.map(_.poolNames).getOrElse(Set(""))
+      } else {
+        Set(IpmiInfo.AddressConfig.flatMap(_.defaultPoolName).getOrElse(""))
```

Comment:
Yeah if no pools are defined in the configuration it trys to pull back the DEFAULT pool. The get or else was just because this returns an option and I felt dirty just using get.